mailpoet/lib/Newsletter/Listing/NewsletterListingRepository.php

```diff
@@ -204,11 +204,21 @@ protected function applyGroup(QueryBuilder $queryBuilder, string $group) {
       ->setParameter('status', $group);
   }
 
-  protected function applySearch(QueryBuilder $queryBuilder, string $search) {
+  protected function applySearch(QueryBuilder $queryBuilder, string $search, array $parameters = []) {
     $search = Helpers::escapeSearch($search);
-    $queryBuilder
-      ->andWhere('n.subject LIKE :search')
-      ->setParameter('search', "%$search%");
+
+    $type = $parameters['type'] ?? null;
+
+    if ($type && $type === NewsletterEntity::TYPE_NOTIFICATION_HISTORY) {
+      $queryBuilder
+        ->join('n.queues', 'sq')
+        ->andWhere('sq.newsletterRenderedSubject LIKE :search')
+        ->setParameter('search', "%$search%");
```

Comment:
Currently [when we create a post notification](https://github.com/mailpoet/mailpoet/blob/c0ef898dc04949d0928b1e353c32be9ec31191ca/mailpoet/lib/Cron/Workers/Scheduler.php#L394-L427) it is set to sending status, but at this time there is no newsletterRenderedSubject or queue. The time depends on the background worker setup. So I would suggest to cover both in the search.

Something like this should work (I haven't tested it):
```php
$queryBuilder
        ->join('n.queues', 'sq')
        ->andWhere('sq.newsletterRenderedSubject LIKE :search OR n.subject LIKE :search')
        ->setParameter('search', "%$search%");
```

Reply:
This is already implemented here https://github.com/mailpoet/mailpoet/blob/d6c953aacec39b70f98735e9edafdcad04a016d7/mailpoet/lib/Newsletter/Listing/NewsletterListingRepository.php#L212-L216

Reply:
Aha! Sorry, I missed that. It was changed later! 👍